templates/elasticsearch.yaml

```diff
@@ -25,18 +25,18 @@ spec:
         resources:
           # need more cpu upon initialization, therefore burstable class
           limits:
-            cpu: 1000m
+            cpu: 2000m
             memory: 2048Mi
           requests:
             cpu: 250m
             memory: 1536Mi
       data:
-        heapSize: 1536m
+        heapSize: 3072m
         resources:
           # need more cpu upon initialization, therefore burstable class
           limits:
-            cpu: 1000m
-            memory: 4096Mi
+            cpu: 4000m
+            memory: 8192Mi
           requests:
             cpu: 250m
-            memory: 2048Mi
+            memory: 8000Mi
```

Comment:
I feel we should simply disallow bursting for memory. Memory bursting relies on OOM score adjustment, which is not very reliable. I would rather prefer a more deterministic behavior on this.

i.e., change this to 8192Mi

Reply:
Disallowing bursting goes against the settings in our ES installations for Kubernetes which recommend
```
In terms of Memory resources you should make sure that you follow that equation:

${role}HeapSize < ${role}MemoryRequests < ${role}MemoryLimits
```

Reply:
For instance in: https://github.com/helm/charts/tree/master/stable/elasticsearch

Reply:
`8000` seems very high when our heapsize is `3072`, could we not adjust our request down to `4096` to significantly improve the pods ability to be scheduled?

Also it does seem like we would want to keep bursting available, as that will provide memory wiggle-room while still giving us a better baseline to get scheduled.

Reply:
OK, that's fair. I remember that we saw memory usage spike to 7G+ in soak. So the limit has to be 8G. The request can be lower.

I think it's OK to reduce memory request to 4608 (1.5x the heap size) given the heap size is 3072.

Thoughts?

Reply:
Sounds good :+1:

Reply:
Done

Reply:
IMHO, we won't see an impact until we start using it, and see if that value is enough or not